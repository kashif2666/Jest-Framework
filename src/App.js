import { useState } from "react";
import "./App.css";
import Users from "./Users";
import handleOtherButton from "./helper";

function App() {
  const [data, setData] = useState("");
  const [data2, setData2] = useState("");
  const handleButton = () => {
    setData2("Hello");
  };

  return (
    <div className="App">
      <p>First Jest Test Case</p>
      <p>Aman Anwar</p>
      {/* <input
        type="text"
        placeholder="Enter username"
        name="username"
        id="userId"
      /> */}
      <br />
      <br />
      {/* <h2>Test onChange event with Input text</h2> */}
      {/* <input
        type="text"
        onChange={(e) => setData(e.target.value + "test")}
        value={data}
      /> */}
      <br /> <br />
      <h2>Test Button Click</h2>
      <button onClick={() => setData("hii")}>Updated Data</button>
      <h2>{data}</h2>
      <br />
      <Users />
      <br /> <br />
      <h2>Functional Component method Testing</h2>
      <button data-testid="btn1" onClick={handleButton}>
        Updated
      </button>
      <button onClick={handleOtherButton}>otherButton</button>
      <h2>{data2}</h2>
      <br />
      <br />
      <img
        title="Man In Weed"
        src="https://plus.unsplash.com/premium_photo-1725024141611-52f6d3870816?q=80&w=1374&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
      />
    </div>
  );
}

export default App;
