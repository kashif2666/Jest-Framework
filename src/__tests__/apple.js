import { fireEvent, render, screen } from "@testing-library/react";
import App from "../App";
import cleanDb from "../service";

// beforeAll(() => {
//   console.log("-----------beforeAll-----------");
// });

// beforeEach(() => {
//   console.log("-----------beforeEach-----------");
// });
test.only("123 First React app case", () => {
  render(<App />);
  const text = screen.getByText(/First Jest Test case/i);
  const text2 = screen.getByText(/Aman Anwar/i);
  const title = screen.getByTitle("Man In Weed");
  expect(text).toBeInTheDocument();
  expect(text2).toBeInTheDocument();
  expect(title).toBeInTheDocument();
});

test("Testing Input Box", () => {
  render(<App />);
  const checkInput = screen.getByRole("textbox");
  const checkInputPlaceholder = screen.getByPlaceholderText("Enter username");
  expect(checkInput).toBeInTheDocument();
  expect(checkInputPlaceholder).toBeInTheDocument();
  expect(checkInput).toHaveAttribute("name", "username");
  expect(checkInput).toHaveAttribute("id", "userId");
  expect(checkInput).toHaveAttribute("type", "text");
});

// grouping using describe, skip, only describe and nested describe

describe("UI test case group", () => {
  test("case 1", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  test("case 2", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  test("case 3", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });
});

// describe.only("API test case group", () => {
//   test("api case 1", () => {
//     render(<App />);
//     let checkInput = screen.getByRole("textbox");
//     expect(checkInput).toHaveAttribute("name", "username");
//   });

//   test("api case 2", () => {
//     render(<App />);
//     let checkInput = screen.getByRole("textbox");
//     expect(checkInput).toHaveAttribute("name", "username");
//   });

//   test("api case 3", () => {
//     render(<App />);
//     let checkInput = screen.getByRole("textbox");
//     expect(checkInput).toHaveAttribute("name", "username");
//   });
// });

describe.skip("API test case group", () => {
  test("api case 1", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  test("api case 2", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  test("api case 3", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });
});

describe("Nested API test case group", () => {
  test("api case 1", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  test("api case 2", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  test("api case 3", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    expect(checkInput).toHaveAttribute("name", "username");
  });

  describe("inner describe", () => {
    test("inner test case", () => {
      render(<App />);
      let checkInput = screen.getByRole("textbox");
      expect(checkInput).toHaveAttribute("name", "username");
    });
  });
});

test("On Change event testing", () => {
  render(<App />);
  let input = screen.getByRole("textbox");
  fireEvent.change(input, { target: { value: "a" } });
  expect(input.value).toBe("atest");
});

test("On Click event testing", () => {
  render(<App />);
  let btn = screen.getByRole("button");
  fireEvent.click(btn);
  expect(screen.getByText("hello")).toBeInTheDocument();
});

test("snapshot for app component", () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});

// afterAll(() => {
//   cleanDb();
// });

// afterEach(() => {
//   console.log("-----------AfterEach-----------");
// });
