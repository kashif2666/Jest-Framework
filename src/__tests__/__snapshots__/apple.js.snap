// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <p>
          First Jest Test Case
        </p>
        <p>
          Aman Anwar
        </p>
        <br />
        <br />
        <br />
         
        <br />
        <h2>
          Test Button Click
        </h2>
        <button>
          Updated Data
        </button>
        <h2 />
        <br />
         
        <br />
        <img
          src="https://plus.unsplash.com/premium_photo-1725024141611-52f6d3870816?q=80&w=1374&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
          title="Man In Weed"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <p>
        First Jest Test Case
      </p>
      <p>
        Aman Anwar
      </p>
      <br />
      <br />
      <br />
       
      <br />
      <h2>
        Test Button Click
      </h2>
      <button>
        Updated Data
      </button>
      <h2 />
      <br />
       
      <br />
      <img
        src="https://plus.unsplash.com/premium_photo-1725024141611-52f6d3870816?q=80&w=1374&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
        title="Man In Weed"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
